lib/Commands/NinjaBuildCommand.cpp

```diff
@@ -61,10 +61,16 @@ static uint64_t getTimeInMicroseconds() {
 }
 
 static std::string getFormattedString(const char* fmt, va_list ap) {
-  char* buf;
-  if (::vasprintf(&buf, fmt, ap) < 0) {
+  int count = ::vsnprintf(NULL, 0, fmt, ap);
+  if (count < 0) {
     return "unable to format message";
   }
+
+  char *buf = static_cast<char *>(malloc(count + 1));
+  if (::vsnprintf(buf, sizeof(buf), fmt, ap) < 0) {
```

Comment:
Jumping in as an outside person interested in llbuild: you could use `std::string` directly as the buffer here, since it's not coming from `asprintf()` any more.